src/client/datascience/ipywidgets/ipyWidgetMessageDispatcher.ts

```diff
@@ -275,12 +275,22 @@ export class IPyWidgetMessageDispatcher implements IIPyWidgetMessageDispatcher {
         let message;
 
         if (!this.isUsingIPyWidgets) {
-            if (!message) {
+            // Lets deserialize only if we know we have a potential case
+            // where this message contains some data we're interested in.
+            let mustDeserialize = false;
+            if (typeof data === 'string') {
+                mustDeserialize = data.includes(WIDGET_MIMETYPE) || data.includes(Identifiers.DefaultCommTarget);
```

Comment:
This is a pretty big assumption. How do you know that all messages for a widget will fit into these subcategories? And what about when we start supporting synapse?

What's the perf improvement of this change? Seems risky to me.

Reply:
Misread the section below. THis is just like the previous change.